be/src/env/s3_client.h

```diff
@@ -0,0 +1,87 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2022-present, StarRocks Limited.
+
+#pragma once
+
+#include <aws/s3/S3Client.h>
+#include <aws/transfer/TransferManager.h>
+
+#include <memory>
+
+#include "env/cloud_storage_client.h"
+
+namespace starrocks {
+
+class S3Credential {
+public:
+    std::string access_key_id;
+    std::string secret_access_key;
+};
+
+class S3Client : public CloudStorageClient {
+public:
+    S3Client(const Aws::Client::ClientConfiguration& config, const S3Credential& cred = S3Credential(),
+             bool use_transfer_manager = false);
+    ~S3Client();
+
+    /*
+     *  Bucket Operation
+     */
+    virtual Status create_bucket(const std::string& bucket_name) override;
+
+    virtual Status delete_bucket(const std::string& bucket_name) override;
+
+    /*
+     *  Object Operation
+     */
+    virtual Status put_object(const std::string& bucket_name, const std::string& object_key,
+                              const std::string& object_path) override;
+
+    virtual Status put_string_object(const std::string& bucket_name, const std::string& object_key,
+                                     const std::string& object_value) override;
+
+    virtual Status get_object(const std::string& bucket_name, const std::string& object_key,
+                              const std::string& object_path) override;
+
+    virtual Status get_object_range(const std::string& bucket_name, const std::string& object_key,
+                                    std::string* object_value, size_t offset, size_t length,
+                                    size_t* read_bytes) override;
+
+    virtual Status get_object_range(const std::string& bucket_name, const std::string& object_key, char* object_value,
+                                    size_t offset, size_t length, size_t* read_bytes) override;
+
+    virtual Status exist_object(const std::string& bucket_name, const std::string& object_key) override;
+
+    virtual Status get_object_size(const std::string& bucket_name, const std::string& object_key,
+                                   size_t* size) override;
+
+    virtual Status delete_object(const std::string& bucket_name, const std::string& object_key) override;
+
+    virtual Status list_objects(const std::string& bucket_name, const std::string& object_prefix,
+                                std::vector<std::string>* result) override;
+
+private:
+    // transfer manager's thread pool.
+    static const int _thread_pool_thread_number = 16;
```

Comment:
shoud start with `k`, refers to https://starrocks.feishu.cn/docs/doccnH0eI3LRYUa40tsEZjviI3c#JJSjdN